ros_ws/src/autonomous/src/emergency_stop.cpp

```diff
@@ -10,32 +10,32 @@ EmergencyStop::EmergencyStop()
 bool EmergencyStop::emergencyStop(const sensor_msgs::LaserScan::ConstPtr& lidar)
 {
     int available_samples = (lidar->angle_max - lidar->angle_min) / lidar->angle_increment;
-    int samples_used = 0;
-    float range_sum = 0;
 
-    int index_start = available_samples / 2 - SAMPLE_ANGLE / lidar->angle_increment / 2;
-    int index_end = available_samples / 2 + SAMPLE_ANGLE / lidar->angle_increment / 2;
+    // calculate the sample_angle of the lidar to be used to check if there is
+    // an obstacle in front of the car, with respect to the RANGE_THRESHOLD
+    float sample_angle = (360 * CAR_BUMPER_LENGTH) / (M_PI * RANGE_THRESHOLD * 2) * DEG_TO_RAD;
+
+    int index_start = available_samples / 2 - sample_angle / lidar->angle_increment / 2;
+    int index_end = available_samples / 2 + sample_angle / lidar->angle_increment / 2;
     ROS_ASSERT_MSG(index_start >= 0 && index_end < available_samples,
-                   "SAMPLE_ANGLE is too big. Trying to access lidar samples out of bounds.");
+                   "sample_angle is too big. Trying to access lidar samples out of bounds.");  
 
+    // determine the minimum distance between the car and a potetial obstacle
+    // Instead of calculating the range average, we are more cautious because we would 
+    // rather have false positives instead of false negatives.
+    // i.e. we would rather stop too much than crash into an obstacle.
+    float min_dist = MAX_RANGE;
     for (int i = index_start; i < index_end; i++)
```

Comment:
If all you're doing is finding the smalles number in the array `lidar->ranges` you could use 
[`std::min_element`](https://en.cppreference.com/w/cpp/algorithm/min_element) from `<algorithm>` to reduce the LoC.

C++11 Example
```C++
auto min_dist = std::min(MAX_RANGE, std::min_element(lidar->ranges + index_start, lidar->ranges + index_end));
```
C++17 Example (using the Parallelism TS)
```C++
auto  min_dist = std::min(std::execution::par, MAX_RANGE, std::min_element(lidar->ranges + index_start, lidar->ranges + index_end));
```